### PR TITLE
Document tests

### PR DESCRIPTION
**Pull request by huddlej**
Adds [documentation about how to run unit tests and doctests](https://github.com/nextstrain/augur/tree/document-tests/docs#tests)

Part of getting doctests to work within the [pytest](https://docs.pytest.org/en/latest/) environment requires temporarily installing augur as a package. I've added a minimal `setup.py` file to the top level for this purpose, but installing augur as a proper module is part of a bigger discussion and PR that involves reorganizing the repository (e.g., to organize code into a module like [treetime](https://github.com/neherlab/treetime) is).

### CODE REVIEW

**Comment by trvrb** [comment]
@huddlej ---

Thanks for doing this. I don't quite understand why it's necessary to install as a package to use PyTest, but more so, I don't understand the harm installing as a package. Why install as package, run PyTest and then uninstall? This seems very clunky. Tests need to be easy to run if people are going to run them.

**Comment by huddlej** [comment]
The installation is required because of how pytest runs doctests. Pytest searches for all .py files with doctests to run, but it doesn't know which package those files are in, so it can't handle imports like `from base import sequences_process`.

Unit tests aren't affected by this issue because you can import the code you want to test with relative imports like `from ..base import sequences_process`.

It looks like [tox](http://tox.readthedocs.io/en/latest/) was designed to get around these issues by creating your package's environment on the fly in a virtualenv, running your tests, and then cleaning up afterwards. This way, all you have to do is run `tox` from the augur base directory and it just works.

Using tox adds a 4 line config file, `tox.ini`, and an additional dependency for augur. I just made an additional commit that adds these pieces. I'll test it out some more and then add documentation.

**Comment by huddlej** [comment]
The following works well on rhino for me with the `janus_python2` environment. @trvrb, would you be able to try this out on rhino and/or your laptop to see if it works?

```bash
# Get augur.
git clone https://github.com/nextstrain/augur.git
cd augur
git checkout document-tests

# Install test tools with conda.
source activate janus_python2
conda install pytest virtualenv
pip install tox

# Run tests.
tox
```

Barney has tried these tests on his laptop and there is an issue with matplotlib when the backend is "MacOSX". I don't see this issue on my laptop with the "Agg" backend though.

**Comment by trvrb** [comment]
Thanks so much for detailed PR @huddlej.

I was getting the same issues from matplotlib. I fixed them with: https://stackoverflow.com/a/21789908 to change the `backend` to `TkAgg`. I don't really understand the difference between these different backends. This feels like a separate issue. I'll make a note about the backend thing, but otherwise this looks great.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[tox]
+envlist = py27
+
+[testenv]
+deps = -rrequirements.txt
```
Comment (trvrb) [inline tox.ini]: @huddlej --- Is this correct? It shouldn't be `deps = -r requirements.txt`?

**Pull request merged**